read-write/README.md

```diff
@@ -0,0 +1,83 @@
+# TypeDB read-write benchmark
+This benchmark contains a 'minimal set of representative queries' to emulate a typical typedb workload.
+It was developed to evaluate the effects of replacing RocksDB with SpeeDB on read & write performance.
+
+### Requirements 
+* Bazel v5.1.1
+* A running [typedb server](https://github.com/vaticle/typedb) to benchmark against.
+
+### Running
+The easiest way to run the benchmark is through bazel:
+
+```bash
+bazel run //read-write:run -- --database=typedb --address=127.0.0.1:1729 --config=read-write/config/<config-file>.yml
+```
+
+Replace the address if you're not running the server on the same machine or using TypeDB's default port. 
+
+The <config-file> determines which queries will be run.
+* `ci-tests.yml`: Runs a few iterations to ensure everything is working properly (< 1 GB)
+* `read-write-benchmark.yml`: Performs (all) read & write queries described below (~10 GB)
+* `write-benchmark.yml`: Only performs write queries described below (~350 GB)
+* `large-data-benchmark.yml`: Performs all queries except `readAddressFromPostCode` for a lage number of iterations (~350 GB)
+
+### Benchmark
+
+The `PersonAgent` currently implements 6 queries.
+* `createPerson`:
+  * Inserts a `person` entity with 3 attributes - `name`, `address`, and `postCode`.
+* `createFriendship`:
+  * selects two names at random (on the client-side)
+  * runs a 'match-insert' query to create a `friendship` relation between them, with a `meeting-time` attribute attached.
+* `readAddressFromName`: 
+  * selects a name at random (client-side)
+  * finds the person with the name
+  * then finds their address. 
+* `readFriendsOf`: pure-read
+  * selects a name at random (client-side)
+  * finds the person with the name
+  * then finds all friendships and the friend (other person) in each friendship
+* `readFriendsOfFriends`: pure-read
+  * selects a name at random (client-side)
+  * finds the person with the name
+  * then finds __all__ friendships and the friend in each friendship
+  * for each such 'other person', it finds __all__ their friendship relations and the associated friends. 
+* `readPersonsByPostCode`: pure-read
+  * selects a postCode at random (client-side)
+  * finds __all__ persons with the associated postCode.
+  * __Note__: Since the number of postCodes is fixed, this query scales linearly as more persons are inserted
```

Comment:
add a section `## Benchmark Architecture`

Let's add a section here which should outline how the benchmark operates in terms of multithreading, determinism, etc. (basically what are the partitions & how is parallelism achieved in relation to the above queries) This helps the new reader understand the some of the keywords such as 'partition' and whether things are deterministic or not, and how that's achieved.

Reply:
Shall we do this under framework and link?